pandas/core/algorithms.py

```diff
@@ -446,7 +447,11 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> npt.NDArray[np.bool_]:
         )
 
     if not isinstance(values, (ABCIndex, ABCSeries, ABCExtensionArray, np.ndarray)):
-        values = _ensure_arraylike(list(values))
+        if not is_signed_integer_dtype(comps):
+            # GH#46485 Use object to avoid upcast to float64 later
```

Comment:
IIUC the problem is that this comes back with int64, then np.find_common_type for int64+uint64 is float64, which is lossy for big integers.

we face a similar problem elsewhere, and ideally id like to re-use some of the logic we use for those.  the place that comes to mind is in Index._find_common_type_compat.

Reply:
So we should move the logic from ``_find_common_type_compat`` into a helper function we can call in both places?

Reply:
if it can be done gracefully thatd be nice.  otherwise i guess a TODO to look into sharing sooner or later